src/Symfony/Component/Console/Helper/Dumper.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Dumper\CliDumper;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class Dumper
+{
+    private $output;
+    private $handler;
+
+    public function __construct(OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->handler = function ($variable): string {
+            switch (true) {
+                case class_exists(CliDumper::class):
```

Comment:
I'd suggest doing this check outside of the closure, and create two separate closures based on the result.
$cloner and $dumper could then be created outside of them also, directly in the constructor.